docs_src/design/adr/security/tbd-authentication-e2ee.md

```diff
@@ -0,0 +1,414 @@
+and OpenZiti Tunneler are the new OpenZiti components.
+Two of these components, the controller and the router,
+are in a dashed box, denoting that these components
+have open ports on the network underlay.
+The controller and the router could be Internet-accessible,
+on a local docker network, or on the host network.
+To permit off-host access to the zero-trust network,
+either the OpenZiti router must be exposed externally on the host,
+or there must be a connection to a peer router that
+is itself accessible off-host.
+"Zitified" clients can simply connect to an OpenZiti edge router
+and join the zero-trust network.
+Non-ziti-aware components such as browsers or command-line clients
+must run on a host with an OpenZiti tunneler
+that works similar to an SSH port-forwarder to
+forward connections onto the zero-trust network.
+There is some security risk in doing this,
+as the OpenZiti Tunneler can't tell the difference between
+a non-zitified client and malware running on the host.
+
+The green boxes,
+showing Vault, Redis, and Consul,
+are existing third-party non-zitified components
+that are already used by EdgeX.
+These components have independent authentication mechanisms.
+They are safe to use for kernel-mediated same-host access,
+but not safe to access over the network without
+additional network security.
+The OpenZiti Router also has an outbound tunneler built-in,
+allowing traffic to traverse the zero-trust network encrypted,
+and exit unencrypted to communicate with services
+listening on the docker network (zero-trust network access, ZTNA),
+or listening on the host network (zero-trust host access, ZTHA).
+These connections are indicated by the blue-violet arrows.
+Note that Vault also has a thick dashed border,
+indicating that it also must have an exposed listening port.
+This is because a Vault-based identity claim is required
+to join the zero-trust network in the normal case,
+and thus Vault must be available as a prerequisite
+to the zero-trust network being available.
+In EdgeX 3.0,
+Vault is exposed behind a TLS-enabled NGINX API gateway (not shown),
+providing confidentiality and integrity protection for those connections.
+
+Connections to exposed ports on the network underlay
+are denoted with red arrows:
+they always terminate at the boxes with thick dashed borders.
+Vault is special in that it also has an alias
+on the zero-trust network to service internal requests
+to the key-value secret store and the consul secrets engine.
+
+The gray boxes in the "zero-trust network" box are
+the Zitified EdgeX services.
+One might assume that connections between these services are peer-to-peer,
+but this is not the case.
+Instead, EdgeX services connect to the OpenZiti edge router
+which perform packet routing functions.
+The edge router applies network policies to the traffic,
+controlling who can speak to whom,
+but payloads are encrypted end-to-end.
+Only the receiving microservice can decrypt the payload,
```

Comment:
> Only the receiving microservice can decrypt the payload,
and the receiver has cryptographic proof of the sending service's identity.

I would reword this. Technically, when the initiating identity connects to the microservice, a key exchange occurs. Every connection is a new key exchange. So it's true that only the receiving microservice can decrypt the payload, but it's arguably not proving the identity of the sender. if that makes sense?

Reply:
Yup.

Reply:
Updated.